http/src/request/application/command/create_guild_command/mod.rs

```diff
@@ -0,0 +1,99 @@
+mod chat_input;
+mod message;
+mod user;
+
+pub use self::{
+    chat_input::CreateGuildChatInputCommand, message::CreateGuildMessageCommand,
+    user::CreateGuildUserCommand,
+};
+
+use super::super::{InteractionError, InteractionErrorType};
+use crate::{request::validate, Client};
+use twilight_model::id::{ApplicationId, GuildId};
+
+/// Create a new command in a guild.
+///
+/// The name must be between 1 and 32 characters in length. Creating a guild
+/// command with the same name as an already-existing guild command in the same
+/// guild will overwrite the old command. See [the discord docs] for more
+/// information.
+///
+/// [the discord docs]: https://discord.com/developers/docs/interactions/slash-commands#create-guild-application-command
+#[must_use = "the command must have a type"]
+pub struct CreateGuildCommand<'a> {
+    application_id: ApplicationId,
+    guild_id: GuildId,
+    http: &'a Client,
+    name: &'a str,
+}
+
+impl<'a> CreateGuildCommand<'a> {
+    pub(crate) fn new(
+        http: &'a Client,
+        application_id: ApplicationId,
+        guild_id: GuildId,
+        name: &'a str,
+    ) -> Result<Self, InteractionError> {
+        if !validate::command_name(name) {
+            return Err(InteractionError {
+                kind: InteractionErrorType::CommandNameValidationFailed,
+            });
+        }
+
+        Ok(Self {
+            application_id,
+            guild_id,
+            http,
+            name,
+        })
+    }
+
+    /// Create a chat input command in a guild.
+    ///
+    /// The description must be between 1 and 100 characters in length. Creating
+    /// a guild command with the same name as an already-existing guild command
+    /// in the same guild will overwrite the old command. See [the discord docs]
+    /// for more information.
+    ///
+    /// # Errors
+    ///
+    /// Returns an [`InteractionErrorType::CommandDescriptionValidationFailed`]
+    /// error type if the command description is not between 1 and
+    /// 100 characters.
+    ///
+    /// [the discord docs]: https://discord.com/developers/docs/interactions/slash-commands#create-guild-application-command
```

Comment:
For all the doc links, they need to be updated to something like https://discord.com/developers/docs/interactions/application-commands#create-guild-application-command as they have moved on the docs site